Detections/ASimDNS/imDns_DomainEntity_DnsEvents.yaml

```diff
@@ -0,0 +1,66 @@
+id: 999e9f5d-db4a-4b07-a206-29c4e667b7e8
+name: (Preview) TI map Domain entity to Dns Event (Normalized DNS)
+description: |
+  'Identifies a match in DNS events from any Domain IOC from TI
+  To use this analytics rule, make sure you have deployed the [ASIM normalization parsers](https://aka.ms/AzSentinelDns)'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: ThreatIntelligence
+    dataTypes:
+      - ThreatIntelligenceIndicator
+  - connectorId: ThreatIntelligenceTaxii
+    dataTypes:
+      - ThreatIntelligenceIndicator
+queryFrequency: 1h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
+tags:
+  - Id: 85aca4d1-5d15-4001-abd9-acb86ca1786a
+    version: 1.0.0
+  - Schema: ASIMDns
+    SchemaVersion: 0.1.1
+query: |
+  let dt_lookBack = 1h;
+  let ioc_lookBack = 14d;
+  //Create a list of TLDs in our threat feed for later validation
+  let list_tlds = ThreatIntelligenceIndicator
+    | where TimeGenerated > ago(ioc_lookBack)
+    | where isnotempty(DomainName)
+    | extend tld = split(DomainName, '.')[-1]
+    | summarize make_set(tld);
+  ThreatIntelligenceIndicator
+    | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
+    | where Active == true
+    // Picking up only IOC's that contain the entities we want
+    | where isnotempty(DomainName)
+    | join (
+        imDns(starttime=ago(dt_lookBack))
+        //Extract domain patterns from syslog message
+        | where isnotempty(DnsQuery)
+        //Split out the TLD
+        | extend parts = split(DnsQuery, '.')
+        //Validate parsed domain by checking if the TLD is in the list of TLDs in our threat feed
+        | extend tld = split(DnsQuery, '.')[-1]
+        | extend DNS_TimeGenerated = TimeGenerated
+  ) on $left.DomainName==$right.DnsQuery
+  | where DNS_TimeGenerated >= TimeGenerated and DNS_TimeGenerated < ExpirationDateTime
+  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+  | project LatestIndicatorTime, Description, ActivityGroupNames, IndicatorId, ThreatType, ExpirationDateTime, ConfidenceScore, Url, DNS_TimeGenerated, Dvc, SrcIpAddr, DnsQuery, QueryType
+  | extend timestamp = DNS_TimeGenerated, HostCustomEntity = Dvc, IPCustomEntity = SrcIpAddr, URLCustomEntity = Url
+entityMappings:
```

Comment:
Let's add entities, custom fields and dynamic values while at it. The first two are easy. The latest we should discuss.